### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by hhamud**
## Summary of Changes

This pull request introduces several improvements to the project:

- **Code Refactoring**: Significant refactoring of the codebase to enhance readability and maintainability.

- **Configuration Updates**: Made changes to the configuration setup for increased flexibility and to address specific issues.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,16 +1,12 @@
 [workspace]
 members = [
-    "lib/cli",
-    "lib/common",
-    "lib/rpc",
-    "lib/hub",
-    "lib/eth",
-    "lib/storage",
+"crates/*"
```
Comment (avichalp) [inline Cargo.toml:3]: nice!

File: README.md [hunk 2]
```diff
@@ -77,3 +77,16 @@ make db-migrate
...
+
+## Contributing
+
+Make sure to run the following command before pushing commits as this will run all the checks needed to pass the CI workflow.
+
+``` bash
+make verify
+```
+
+
```
Comment (avichalp) [inline README.md:91]: nit: extra new lines

File: crates/crypto/src/eip712.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    i64::deserialize(deserializer).map(U256::from)
+}
+
+#[derive(Clone, Debug, Deserialize, Eip712, EthAbiType)]
+#[eip712(
+    name = "Farcaster Verify Ethereum Address",
+    version = "2.0.0",
+    salt = "0xf2d857f4a3edcb9b78b4d503bfe733db1e3f6cdc2b7971ee739626c97e86a558"
+)]
+
```
Comment (avichalp) [inline crates/crypto/src/eip712.rs:18]: Why is there a new line between macros?

File: crates/crypto/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,5 @@
+pub mod blake3;
+pub mod ed25519;
+//pub mod eip712;
+//pub mod signers;
```
Comment (avichalp) [inline crates/crypto/src/lib.rs:4]: maybe we can remove the commented modules?

File: crates/hub/src/bin/main.rs [hunk 5]
```diff
@@ -0,0 +1,19 @@
+use clap::Parser;
+use teleport_cli::{Cli, Commands};
+use teleport_hub::Hub;
+
+type HResult<T> = Result<T, Box<dyn std::error::Error>>;
```
Comment (avichalp) [inline crates/hub/src/bin/main.rs:5]: should we use something like `anyhow`?

File: crates/crypto/src/lib.rs (hunk 4 above)
Comment (hhamud) [reply crates/crypto/src/lib.rs:4]: Good point.

File: crates/hub/src/bin/main.rs (hunk 5 above)
Comment (hhamud) [reply crates/hub/src/bin/main.rs:5]: that's a better way of doing this.

**Review by avichalp — APPROVED** [review]
Thank you @hhamud for taking the time to clean things up here. I believe, there is a lot of cleanup left to do in this repo but I will merge this PR for now.

**Pull request merged**